x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/jdbc/TypeConverter.java

```diff
@@ -94,15 +115,34 @@ static Timestamp convertTimestamp(Long millis, Calendar cal) {
             c.setTimeInMillis(initial);
         }
     }
+    
+    static long convertFromCalendarToUTC(long value, Calendar cal) {
+        if (cal == null) {
+            return value;
+        }
+        Calendar c = (Calendar) cal.clone();
+        c.setTimeInMillis(value);
+
+        ZonedDateTime convertedDateTime = ZonedDateTime
+                .ofInstant(c.toInstant(), ZoneOffset.ofTotalSeconds(c.getTimeZone().getRawOffset() / 1000))
+                .withZoneSameLocal(ZoneOffset.UTC);
+
+        return convertedDateTime.toEpochSecond() * 1000 + convertedDateTime.getNano() / 1_000_000;
+    }
 
     /**
      * Converts object val from columnType to type
      */
     @SuppressWarnings("unchecked")
-    static <T> T convert(Object val, JDBCType columnType, Class<T> type) throws SQLException {
+    static <T> T convert(Object val, JDBCType columnType, Class<T> type) throws SQLException, ClassCastException {
```

Comment:
CCE is never a good idea to throw out - especially since it forces the caller to handle it.
It should be handled inside convert directly.

Reply:
Fixed